lib/pure/collections/sets.nim

```diff
@@ -1020,7 +1020,7 @@ when isMainModule and not defined(release):
 
     block toSeqAndString:
       var a = toHashSet([2, 7, 5])
-      var b = initHashSet[int]()
+      var b = initHashSet[int](rightSize(a.len))
```

Comment:
Why is this needed?

Reply:
so the test passes. fixing the test after a hash algorithm update is not incorrect in this case; the ordering of elements within the HashSet is implementation defined

Reply:
No this is simply incorrect. The initial size is optional.

Reply:
sure but I'm not sure what you're suggesting concretely ; that was the simplest fix here; note that the test is not good anyway, because ordering is implementation defined; so it'll break every time implementation changes; a better test would be to have `toStringSorted(a: HashSet): string` instead of `$a` which would work regardless of implementation

Reply:
```suggestion
      var b = initHashSet[int]()
```
Thats what I am suggesting : )
I was only referring to this particular line

Reply:
Ah, I see what you mean now. But the fact that initializing with a initialSize changes the order this way is also implementaion defined.

Reply:
it's implementation defined but the way i wrote it should work regardless of hash implementation, based on how `toHashSet` is implemented; in any case my main point is that a better test (in general, for any algo testing HashSet, Tables + other structures based on hashes) would involve defining a `itemsSorted(collection)`; left for future PRs